iothub_client/tests/common_device_method_e2e/iothubclient_common_device_method_e2e.c

```diff
@@ -164,61 +164,71 @@ static int MethodCallback(const char* method_name, const unsigned char* payload,
+static void CommandCallback(const IOTHUB_CLIENT_COMMAND_REQUEST* commandRequest, IOTHUB_CLIENT_COMMAND_RESPONSE* commandResponse,  void* userContextCallback)
 {
-    int responseCode;
     const char * expectedMethodPayload = (const char*)userContextCallback;
 
-    LogInfo("CommandCallback invoked with componentName=%s, commandName=%s", componentName, commandName);
+    LogInfo("CommandCallback invoked with componentName=%s, commandName=%s", commandRequest->componentName, commandRequest->commandName);
 
-    if ((componentName == NULL) || (strcmp(TEST_COMMAND_COMPONENT_NAME, componentName) != 0))
+    if ((commandRequest->componentName == NULL) || (strcmp(TEST_COMMAND_COMPONENT_NAME, commandRequest->componentName) != 0))
     {
-        LogError("Component name incorrect - expected %s but got %s", TEST_COMMAND_COMPONENT_NAME, componentName);
-        responseCode = METHOD_RESPONSE_ERROR;
+        LogError("Component name incorrect - expected %s but got %s", TEST_COMMAND_COMPONENT_NAME, commandRequest->componentName);
+        commandResponse->statusCode = METHOD_RESPONSE_ERROR;
     }
-    else if ((commandName == NULL) || (strcmp(TEST_COMMAND_COMMAND_NAME, commandName) != 0))
+    else if ((commandRequest->commandName == NULL) || (strcmp(TEST_COMMAND_COMMAND_NAME, commandRequest->commandName) != 0))
     {
-        LogError("Command name incorrect - expected %s but got %s", TEST_COMMAND_COMMAND_NAME, commandName);
-        responseCode = METHOD_RESPONSE_ERROR;
+        LogError("Command name incorrect - expected %s but got %s", TEST_COMMAND_COMMAND_NAME, commandRequest->commandName);
+        commandResponse->statusCode = METHOD_RESPONSE_ERROR;
+    }
+    else if (IOTHUB_CLIENT_COMMAND_REQUEST_STRUCT_VERSION_1 != commandRequest->structVersion)
+    {
+        LogError("Command request structVersion wrong - expected %d but got %d", IOTHUB_CLIENT_COMMAND_REQUEST_STRUCT_VERSION_1, commandRequest->structVersion);
+        commandResponse->statusCode = METHOD_RESPONSE_ERROR;
     }
-    else if (payloadContentType != NULL)
+    else if (IOTHUB_CLIENT_COMMAND_REQUEST_STRUCT_VERSION_1 != commandResponse->structVersion)
+    {
+        LogError("Command response structVersion wrong - expected %d but got %d", IOTHUB_CLIENT_COMMAND_REQUEST_STRUCT_VERSION_1, commandResponse->structVersion);
+        commandResponse->statusCode = METHOD_RESPONSE_ERROR;
+    }
+    else if (commandRequest->payloadContentType != NULL)
     {
         LogError("Payload content type is non-NULL but should have been NULL");
-        responseCode = METHOD_RESPONSE_ERROR;
+        commandResponse->statusCode = METHOD_RESPONSE_ERROR;
     }
-    else if (size != strlen(expectedMethodPayload))
+    else if (commandRequest->payloadLength != strlen(expectedMethodPayload))
```

Comment:
nit: should we change `expectedMethodPayload` to `expectedCommandPayload`

Reply:
I complained about FreeRTOS doxgen confusing this last week I recall so what goes around comes around....  Fixing.